staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admissionreview.go

```diff
@@ -54,10 +54,10 @@ func createAdmissionReview(attr admission.Attributes) admissionv1alpha1.Admissio
 			SubResource: attr.GetSubresource(),
 			Operation:   admissionv1alpha1.Operation(attr.GetOperation()),
 			Object: runtime.RawExtension{
-				Object: attr.GetObject(),
+				Object: versioned.object,
```

Comment:
I don't object to the solution in this PR but I think we need to ask ourselves why the objects currently in `attr` are in the internal version. Doesn't that break other things, too?

Reply:
I can see two blockers from building `attr` with versioned objects,
1. defaulting is done as part of the conversion, we don want to run admissions after defaulting
2. the static validation is done on the internal objects.

Reply:
1. You mean we *do* want to run admission after defaulting?
2. That doesn't mean you have to pass the internal object through the stack?

Reply:
1. yes. fixed the comment.

The current flow and ways to eliminate extra conversion:
* do defaulting on the external types, but it's invoked during the conversion to the internal version.
* convert to external version, run mutating webhooks, convert back to internal version
  * to stop the conversion here, **we need to move the invocation of defaulting out of the conversion stack**
* do validation on internal versions
* convert to external version, run validating webhooks
  * **to stop the conversion here, we need to do validation on external types.**
* convert to the storage version, store to etcd.